src/test/kotlin/pt/up/fe/ni/website/backend/controller/EventControllerTest.kt

```diff
@@ -725,6 +812,169 @@ internal class EventControllerTest @Autowired constructor(
         }
     }
 
+    @NestedTest
+    @DisplayName("PUT /events/{idEvent}/gallery/addImage")
+    inner class AddGalleryImage {
+
+        private val uuid: UUID = UUID.randomUUID()
+        private val mockedSettings = Mockito.mockStatic(UUID::class.java)
+
+        @BeforeAll
+        fun setupMocks() {
+            Mockito.`when`(UUID.randomUUID()).thenReturn(uuid)
+        }
+
+        @AfterAll
+        fun cleanup() {
+            mockedSettings.close()
+        }
+
+        @BeforeEach
+        fun addToRepositories() {
+            accountRepository.save(testAccount)
+            repository.save(testEvent)
+        }
+
+        @Test
+        fun `should add an image`() {
+            val expectedImagePath = "${uploadConfigProperties.staticServe}/events/gallery/${testEvent.title}-$uuid.jpeg"
+
+            mockMvc.multipartBuilder("/events/${testEvent.id}/gallery/addImage")
+                .asPutMethod()
+                .addFile("image", contentType = MediaType.IMAGE_JPEG_VALUE)
+                .perform()
+                .andExpectAll(
+                    status().isOk,
+                    content().contentType(MediaType.APPLICATION_JSON),
+                    jsonPath("$.title").value(testEvent.title),
+                    jsonPath("$.description").value(testEvent.description),
+                    jsonPath("$.teamMembers.length()").value(testEvent.teamMembers.size),
+                    jsonPath("$.gallery.length()").value(1),
+                    jsonPath("$.gallery[0]").value(expectedImagePath),
```

Comment:
Have you (manually) tested if you can access this path on our machine (e.g. open in your browser)? I assume this was already tested for other images but I don't fully remember

Reply:
It is exists on the filesystem but the url does not work (I don't know if I am missing something because the port is 3000 and there is nothing there).

Reply:
@BrunoRosendo

Reply:
Please check #71 again, I'm pretty sure @DoStini already had the logic for this. There is a folder where Spring will save public files (see https://github.com/NIAEFEUP/website-niaefeup-backend/wiki/File-Upload#static-file-uploader)

I suspect the servePath is not being used correctly

Reply:
Images are being stored locally in the `build/resources/main/static` folder, Spring Boot automatically serves the `src/main/resources/static` folder, however, due to the `endpointNotFound` present on the `ErrorController` (`src/main/kotlin/pt/up/fe/ni/website/backend/controller/ErrorController`) having a `RequestMapping("/**")`, it overrides the default behaviour and responds with an error 404 "invalid endpoint". Altough images are not being served, they also are not in the correct place for it, as they would not be served in the build directory.
To test this behaviour, remove the two `*` in the `RequestMapping` mentioned (leaving only `/`) and create an about.html with a `Hello World` string inside, you will see that it exists in `http://localhost:8080/about.html`.